modules/styling-transform/lib/styleTransform.ts

```diff
@@ -0,0 +1,607 @@
+
+function getStyleValueFromType(node: ts.Node, type: ts.Type, checker: ts.TypeChecker) {
+  const value = getCSSValueAtLocation(node as ts.Expression, checker, type); //?
+  if (value) {
+    if (value.startsWith('--')) {
+      return `var(${value})`;
+    }
+    return value;
+  }
+
+  const typeValue = checker.typeToString(type); //?
+
+  throw new Error(
+    `Unknown type at: "${node.getText()}". Received "${typeValue}"\n${getErrorMessage(
+      node
+    )}\nFor static analysis of styles, please make sure all types resolve to string or numeric literals. Please use 'const' instead of 'let'. If using an object, cast using "as const" or use an interface with string or numeric literals.`
+  );
+}
+
+function getPropertyAccessExpressionText(node: ts.PropertyAccessExpression): string {
+  if (ts.isIdentifier(node.name)) {
+    if (ts.isIdentifier(node.expression)) {
+      return `${node.expression.text}.${node.name.text}`;
+    }
+    if (ts.isPropertyAccessExpression(node.expression)) {
+      return `${getPropertyAccessExpressionText(node.expression)}.${node.name.text}`;
+    }
+  }
+  return '';
+}
+
+function parseStyleObjValue(
+  initializer: ts.Node,
+  variables: Record<string, string>,
+  checker: ts.TypeChecker
+): string {
+  /**
+   * String literals like 'red' or empty Template Expressions like `red`
+   */
+  if (ts.isStringLiteral(initializer) || ts.isNoSubstitutionTemplateLiteral(initializer)) {
+    return initializer.text;
+  }
+
+  // numeric literal values like `12`
+  if (ts.isNumericLiteral(initializer)) {
+    return `${initializer.text}px`;
+  }
+
+  // The source file is using an identifier which will be known at runtime, we'll try to
+  // determine the type
+  if (ts.isIdentifier(initializer)) {
+    const type = checker.getTypeAtLocation(initializer);
+    return getStyleValueFromType(initializer, type, checker);
+  }
+
+  /**
+   * ```ts
+   * PropertyAccessExpressions are dot-notation
+   *
+   * foo.bar.baz
+   * ```
+   */
+  if (ts.isPropertyAccessExpression(initializer)) {
+    const type = checker.getTypeAtLocation(initializer);
+    return getStyleValueFromType(initializer, type, checker);
+  }
+
+  /**
+   * This will find patterns like:
+   *
+   * ```ts
+   * cssVar(myVars.color);
+   * cssVar(myVars.colors.background)
+   * ```
+   */
+  if (
+    ts.isCallExpression(initializer) &&
+    ts.isIdentifier(initializer.expression) &&
+    initializer.expression.text === cssVarExpressionName
+  ) {
+    const value = getCSSValueAtLocation(initializer.arguments[0], checker); //?
+    const value2 = initializer.arguments[1]
+      ? parseStyleObjValue(initializer.arguments[1], variables, checker)
+      : undefined; //?
+
+    // handle fallback variables
+    const fallbackValue = getFallbackVariable(value, variables); //?
+    if (value && (value2 || fallbackValue)) {
+      return `var(${value}, ${
+        (value2?.startsWith('--') ? `var(${value2})` : value2) || fallbackValue
+      })`;
+    }
```

Comment:
JK :) but also if it's not a variable yeah?

Reply:
If it is a variable (detected by the `--`), it will wrap with a `var()`. Otherwise it will leave it alone